### PR TITLE
Improvements from sprint II fixed

### PR DESCRIPTION
**Pull request by Faka-GLB**
### whats has been done
- [x] Location picker UI: 
Replaced UIPicker with a material components textField, upon clicking it, will display a picker with the locations, when selected a location, picker closes and sets the text.

- [x] Button manually background:
Manual button from the camera screen has its corners unrounded, color changed and moved slightly up

- [x] Hint color gray:
Changed the hints label color to gray, when the user taps on a field the label moves to the top of the field and turns black

- [x] Add icon to save button:
This fix was already solved in the previous sprint

- [x] Add necessary unit tests:
Some unit tests were added/modified with the changes to the form screen

Out of scope fixes:
Unit tests were importing the old app name (with capitalised A) and needed to update some variable names and functions parameters
Reduced the space between the save icon and the button text in the form screen
Changed manual button text and icon color to black in the camera screen

### Evidence:

Picker and hints colors
iPhone 11 Pro 
 output-onlinepngtools (43)

Demo:
https://user-images.githubusercontent.com/90403672/151868583-975b46f0-1a2b-4211-b880-6354d090b8dd.mp4

Manual button background color:
iPhone 11 Pro 
 output-onlinepngtools (41)

Save Button
iPhone 11 
 output-onlinepngtools (44)

Tests:
Screen Shot 2022-01-28 at 12 55 06

### CODE REVIEW

File: stockApp/stockApp/Presentation/View/CameraView.swift [hunk 1]
```diff
@@ -2,49 +2,51 @@ import UIKit
...
+	static let cameraHeight: CGFloat = 0.92
+	static let edgeInsetsZero: CGFloat = 0
+	static let imageEdgedInsetRight: CGFloat = 30
+	static let edgeInsetsLeft: CGFloat = 10
+	static let edgeInsetsRight: CGFloat = 10
+	static let titleLabelWidth: CGFloat = 21
+	static let zeroAnchors: CGFloat = 0
 	static let manualButtonCornerRadius: Int = 20
+    static let buttonBottomSpacing: CGFloat = 10
+    static let buttonHeightMultiplier: CGFloat = 0.04
```
Comment (ezequielvelezglobant) [inline stockApp/stockApp/Presentation/View/CameraView.swift:15]: fix alignment

File: stockApp/stockApp/Presentation/View/CameraView.swift [hunk 2]
```diff
@@ -74,45 +76,47 @@ class CameraView: UIView {
...
-            manualInputUIButton.setTitle(Strings.manualInputButtonTextSearchAsset,
+            manualInputButton.setTitle(Strings.manualInputButtonTextSearchAsset,
                                          for: .normal)
         }
-		self.addSubview(cameraUIView)
-		self.addSubview(manualInputUIButton)
+        cameraView.addSubview(manualInputButton)
+        self.addSubview(cameraView)
 		setConstraints()
 	}
```
Comment (ezequielvelezglobant) [inline stockApp/stockApp/Presentation/View/CameraView.swift:88]: fix alignment

File: stockApp/stockApp/Presentation/View/CameraView.swift [hunk 3]
```diff
@@ -74,45 +76,47 @@ class CameraView: UIView {
...
+                multiplier: Constants.buttonHeightMultiplier),
+			manualInputButton.topAnchor.constraint(equalTo: cameraView.bottomAnchor,
+                                                     constant: Constants.zeroAnchors),
+            manualInputButton.leadingAnchor.constraint(equalTo: cameraView.leadingAnchor,
+                                                         constant: Constants.zeroAnchors),
+            manualInputButton.trailingAnchor.constraint(equalTo: cameraView.trailingAnchor,
+                                                          constant: Constants.zeroAnchors),
+            manualInputButton.bottomAnchor.constraint(
+                equalTo: cameraView.bottomAnchor,
+                constant: -Constants.buttonBottomSpacing)
```
Comment (ezequielvelezglobant) [inline stockApp/stockApp/Presentation/View/CameraView.swift:113]: fix alignment

File: stockApp/stockApp/Presentation/View/FormView.swift [hunk 4]
```diff
@@ -36,14 +36,15 @@ private struct Constants {
...
-	static let saveTop: Int = 30
+	static let saveTop: Int = 35
 	static let saveCornerRadius: Int = 22
 	static let updateTitle = "Update asset form"
 	static let pickerComponent = 0
 	static let pickerFirstRow = 0
 	static let picekrSecondRow = 1
     static let updateDescriptionFormUILabelText = "Update the following information"
     static let updateButtonText = "Update"
+    static let locationTextFieldHint = "Location"
```
Comment (ezequielvelezglobant) [inline stockApp/stockApp/Presentation/View/FormView.swift:47]: fix alignment

File: stockApp/stockApp/Presentation/View/FormView.swift [hunk 5]
```diff
@@ -84,60 +85,83 @@ class FormView: UIView {
...
+        textField.label.textColor = UIColor.lightGray
 		textField.label.textAlignment = .center
 		textField.backgroundColor = Color.porcelainWhite
 		textField.sizeToFit()
 		textField.autocapitalizationType = .none
 		textField.rightViewMode = .always
 		textField.rightView = UIImageView(image: UIImage(named: Strings.barcodeSearchIcon))
 		textField.rightView?.tintColor = Color.tundoraBlack
+        textField.setNormalLabelColor(UIColor.gray, for: .normal)
+        textField.setFloatingLabelColor(Color.tundoraBlack, for: .normal)
```
Comment (ezequielvelezglobant) [inline stockApp/stockApp/Presentation/View/FormView.swift:104]: fix alignment

File: stockApp/stockApp/Presentation/View/FormView.swift [hunk 6]
```diff
@@ -84,60 +85,83 @@ class FormView: UIView {
...
 		textField.label.lineBreakMode = .byWordWrapping
 		textField.label.numberOfLines = Constants.numberOfLines
-		textField.label.textColor = Color.tundoraBlack
+        textField.label.textColor = UIColor.lightGray
 		textField.label.textAlignment = .center
 		textField.backgroundColor = Color.porcelainWhite
 		textField.sizeToFit()
 		textField.autocapitalizationType = .none
+        textField.setNormalLabelColor(UIColor.gray, for: .normal)
+        textField.setFloatingLabelColor(Color.tundoraBlack, for: .normal)
```
Comment (ezequielvelezglobant) [inline stockApp/stockApp/Presentation/View/FormView.swift:121]: fix alignment

File: stockApp/stockApp/Presentation/View/FormView.swift [hunk 7]
```diff
@@ -84,60 +85,83 @@ class FormView: UIView {
...
 		return textField
```
Comment (ezequielvelezglobant) [inline stockApp/stockApp/Presentation/View/FormView.swift:139]: fix alignment

File: stockApp/stockApp/Presentation/View/FormView.swift [hunk 8]
```diff
@@ -180,13 +204,14 @@ class FormView: UIView {
...
-		scroll.addSubview(sitePicker)
-		scroll.addSubview(saveUIButton)
+		scroll.addSubview(titleLabel)
+		scroll.addSubview(descriptionLabel)
+		scroll.addSubview(barcodeTextField)
+		scroll.addSubview(assetNameTextField)
+		scroll.addSubview(userMailTextField)
+		scroll.addSubview(locationTextField)
+		scroll.addSubview(saveButton)
+        locationTextField.inputView = locationPicker
```
Comment (ezequielvelezglobant) [inline stockApp/stockApp/Presentation/View/FormView.swift:214]: fix alignment

File: stockApp/stockApp/Presentation/View/FormView.swift [hunk 9]
```diff
@@ -285,21 +310,17 @@ class FormView: UIView {
...
-		} else {
-			sitePicker.selectRow(Constants.picekrSecondRow, inComponent: Constants.pickerComponent, animated: false)
-		}
+		titleLabel.text = Constants.updateTitle
+        descriptionLabel.text = Constants.updateDescriptionFormUILabelText
+        saveButton.setTitle(Constants.updateButtonText, for: .normal)
+		barcodeTextField.text = asset.barcode
+		assetNameTextField.text = asset.assetName
+		userMailTextField.text = asset.userMail
+        locationTextField.text = asset.location
```
Comment (ezequielvelezglobant) [inline stockApp/stockApp/Presentation/View/FormView.swift:323]: fix alignment

File: stockApp/stockAppTests/FormViewTest.swift [hunk 10]
```diff
@@ -16,25 +16,26 @@ class FormViewTest: XCTestCase {
...
-		XCTAssertNotNil(formView.assetNameUITextField)
-		XCTAssertNotNil(formView.globerMailUITextField)
-		XCTAssertNotNil(formView.sitePicker)
-		XCTAssertNotNil(formView.saveUIButton)
+		XCTAssertNotNil(formView.titleLabel)
+		XCTAssertNotNil(formView.descriptionLabel)
+		XCTAssertNotNil(formView.barcodeTextField)
+		XCTAssertNotNil(formView.assetNameTextField)
+		XCTAssertNotNil(formView.userMailTextField)
+        XCTAssertNotNil(formView.locationTextField)
```
Comment (ezequielvelezglobant) [inline stockApp/stockAppTests/FormViewTest.swift:24]: fix alignment

File: stockApp/stockAppTests/FormViewTest.swift [hunk 11]
```diff
@@ -43,13 +44,13 @@ class FormViewTest: XCTestCase {
...
-		XCTAssertEqual(formView.titleUILabel.text, TestConstants.title)
-		XCTAssertEqual(formView.barcodeUITextField.text, TestConstants.barcode)
-		XCTAssertEqual(formView.assetNameUITextField.text, TestConstants.assetName)
-		XCTAssertEqual(formView.globerMailUITextField.text, TestConstants.userMail)
-		XCTAssertEqual(TestConstants.siteArray[formView.sitePicker.selectedRow(inComponent: 0)], TestConstants.siteTandil)
+		XCTAssertEqual(formView.titleLabel.text, TestConstants.title)
+		XCTAssertEqual(formView.barcodeTextField.text, TestConstants.barcode)
+		XCTAssertEqual(formView.assetNameTextField.text, TestConstants.assetName)
+		XCTAssertEqual(formView.userMailTextField.text, TestConstants.userMail)
+        XCTAssertEqual(formView.locationTextField.text, TestConstants.siteTandil)
```
Comment (ezequielvelezglobant) [inline stockApp/stockAppTests/FormViewTest.swift:53]: fix alignment

File: stockApp/stockAppTests/FormViewTest.swift [hunk 12]
```diff
@@ -58,17 +59,17 @@ class FormViewTest: XCTestCase {
 			assetName: TestConstants.assetName,
 			userMail: TestConstants.userMail,
 			recordDateTime: TestConstants.recordDateTime,
-			site: TestConstants.siteOther)
+            location: TestConstants.siteOther)
```
Comment (ezequielvelezglobant) [inline stockApp/stockAppTests/FormViewTest.swift:62]: fix alignment

**Review by ezequielvelezglobant — CHANGES_REQUESTED** [review]

**Review by ezequielsetzesglobant — APPROVED** [review]

File: stockApp/stockAppTests/FormViewTest.swift (hunk 12 above)
Comment (Faka-GLB) [reply stockApp/stockAppTests/FormViewTest.swift:62]: done

File: stockApp/stockAppTests/FormViewTest.swift (hunk 11 above)
Comment (Faka-GLB) [reply stockApp/stockAppTests/FormViewTest.swift:53]: done

File: stockApp/stockAppTests/FormViewTest.swift (hunk 10 above)
Comment (Faka-GLB) [reply stockApp/stockAppTests/FormViewTest.swift:24]: done

File: stockApp/stockApp/Presentation/View/FormView.swift (hunk 9 above)
Comment (Faka-GLB) [reply stockApp/stockApp/Presentation/View/FormView.swift]: done

File: stockApp/stockApp/Presentation/View/FormView.swift (hunk 8 above)
Comment (Faka-GLB) [reply stockApp/stockApp/Presentation/View/FormView.swift:214]: done

File: stockApp/stockApp/Presentation/View/FormView.swift (hunk 5 above)
Comment (Faka-GLB) [reply stockApp/stockApp/Presentation/View/FormView.swift]: done

File: stockApp/stockApp/Presentation/View/FormView.swift (hunk 6 above)
Comment (Faka-GLB) [reply stockApp/stockApp/Presentation/View/FormView.swift]: done

File: stockApp/stockApp/Presentation/View/FormView.swift (hunk 7 above)
Comment (Faka-GLB) [reply stockApp/stockApp/Presentation/View/FormView.swift]: done

File: stockApp/stockApp/Presentation/View/FormView.swift (hunk 4 above)
Comment (Faka-GLB) [reply stockApp/stockApp/Presentation/View/FormView.swift:47]: done

File: stockApp/stockApp/Presentation/View/CameraView.swift (hunk 3 above)
Comment (Faka-GLB) [reply stockApp/stockApp/Presentation/View/CameraView.swift]: done

File: stockApp/stockApp/Presentation/View/CameraView.swift (hunk 2 above)
Comment (Faka-GLB) [reply stockApp/stockApp/Presentation/View/CameraView.swift]: done

File: stockApp/stockApp/Presentation/View/CameraView.swift (hunk 1 above)
Comment (Faka-GLB) [reply stockApp/stockApp/Presentation/View/CameraView.swift:15]: done

**Review by ezequielvelezglobant — APPROVED** [review]

**Pull request merged**